test/functional/tests/memory/test_memory_metadata_consumption.py

```diff
@@ -0,0 +1,200 @@
+        sync()
+        fio = (Fio().create_command()
+               .direct()
+               .io_engine(IoEngine.libaio)
+               .size(cache.cache_device.size)
+               .read_write(ReadWrite.read)
+               .io_depth(64)
+               .block_size(Size(1, Unit.Blocks4096)))
+        for dev in core_list:
+            fio.add_job(f"job_{dev.core_id}").target(dev.system_path)
+        fio.run()
+
+        occupancy = cache.get_statistics(percentage_val=True).usage_stats.occupancy
+        TestRun.LOGGER.info(f"Cache occupancy: {occupancy}")
+        if occupancy < (100 - percentage_tolerance):
+            TestRun.LOGGER.warning(f"Cache occupancy is below expectation.")
+
+    with TestRun.step("Measure allocates size of metadata."):
+        used_memory = get_metadata_size_from_vmallocinfo(cas_cache_module)
+        TestRun.LOGGER.info(f"Memory allocated: {used_memory}")
+
+    with TestRun.step("Compare allocated size with cache usage statistics "
+                      "and required consumption."):
+        free_mem_after_start = get_mem_available_from_meminfo()
+
+        metadata_max_size = cache.get_metadata_max_size(cache_line_size)
+        TestRun.LOGGER.info(f"Metadata max size: {metadata_max_size}")
+
+        validate_memory_consumption(cache, metadata_max_size, used_memory, decupled=True)
+
+        memory_delta = int(free_mem_before_start - free_mem_after_start)
+        ram_change_status = is_memory_within_limit("Measured increment of RAM usage after start",
+                                                   metadata_max_size, memory_delta, decupled=True)
+
+        TestRun.LOGGER.info("Stop CAS and verify memory changes again")
+        casadm.stop_all_caches()
+
+        free_mem_after_stop = get_mem_available_from_meminfo()
+        memory_delta = int(free_mem_after_stop - free_mem_after_start)
+        ram_change_status |= is_memory_within_limit("Measured decrement of RAM usage after stop",
+                                                    metadata_max_size, memory_delta, decupled=True)
+
+        if ram_change_status:
+            TestRun.LOGGER.info(f"Memory measurements completed with correct results.")
+        else:
+            TestRun.LOGGER.error(f"Memory measurements completed, found problems.")
+
+
+def is_memory_within_limit(value_name, expected, used, decupled=False):
+    """Check if size of used memory is within expected maximum size"""
+    if decupled:
+        expected_to_check = expected * 10
+        used_to_check = used * 10
+    else:
+        expected_to_check = expected
+        used_to_check = used
+
+    if expected_to_check >= used_to_check:
+        TestRun.LOGGER.info(f"{value_name} ({used}) is within limit ({expected}).")
+        return True
+    else:
+        TestRun.LOGGER.error(f"{value_name} ({used}) exceeds limit ({expected}).")
+        return False
+
+
+def validate_memory_consumption(cache_device, expected_maximum, actual_size, decupled=False):
```

Comment:
"_decupled_" is never used with the default "_False_" value, so it seems that it's not necessary for both _is_memory_within_limit_ and _validate_memory_consumption_